pandas/io/common.py

```diff
@@ -753,6 +761,30 @@ def get_handle(
                         f"Only one file per ZIP: {zip_names}"
                     )
 
+        # TAR Encoding
+        elif compression == "tar":
+            if "mode" not in compression_args:
+                compression_args["mode"] = ioargs.mode
+            if is_path:
+                handle = _BytesTarFile.open(name=handle, **compression_args)
+            else:
+                handle = _BytesTarFile.open(fileobj=handle, **compression_args)
+            assert isinstance(handle, _BytesTarFile)
+            if handle.mode == "r":
+                handles.append(handle)
+                files = handle.getnames()
+                if len(files) == 1:
+                    file = handle.extractfile(files[0])
+                    assert file is not None
+                    handle = file
+                elif len(files) == 0:
+                    raise ValueError(f"Zero files found in TAR archive {path_or_buf}")
+                else:
+                    raise ValueError(
+                        "Multiple files found in TAR archive. "
+                        f"Only one file per TAR archive: {files}"
```

Comment:
test hits this?

Reply:
same as above

Reply:
added test in https://github.com/pandas-dev/pandas/pull/44787/commits/086c59883e169fe021873f25304e7bb09f2823fd